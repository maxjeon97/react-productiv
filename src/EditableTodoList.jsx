import React from "react";
import EditableTodo from "./EditableTodo";

/** Show list of editable todos.
 *
 * Props:
 * - todos: array of [ todo, ... ]
 * - update(): fn to call to update a todo
 * - remove(): fn to call to remove a todo
 * - handleTitleClick(): fn to toggle isDone property of Todo
 *
 * TodoApp -> EditableTodoList -> [ EditableTodo, ... ]
 */

function EditableTodoList({ todos, update, remove, handleTitleClick }) {
  return (
    <div className="EditableTodoList">
      {todos.map(todo => <EditableTodo
        key={todo.id}
        todo={todo}
        handleTitleClick={handleTitleClick}
        update={update}
        remove={remove} />)}
    </div>
  );
}

export default EditableTodoList;
