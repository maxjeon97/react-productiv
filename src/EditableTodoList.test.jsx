import { describe, test, expect } from "vitest";

import React from "react";
import { render } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList";

const testTodos = [
    {
        id: 1,
        title: "Test1",
        description: "Test1 Descr",
        priority: 1,
        isDone: false,
    },
    {
        id: 2,
        title: "Test2",
        description: "Test2 Descr",
        priority: 2,
        isDone: false,
    },
];

describe("EditableTodoList component", function () {
    test("renders without crashing", function () {
        render(<EditableTodoList todos={testTodos} />);
    });

    test('matches snapshot', function () {
        const { container } = render(<EditableTodoList todos={testTodos} />);
        expect(container).toMatchSnapshot();
    });

    test("contains the expected number of EditableTodos", function () {
        const result = render(<EditableTodoList todos={testTodos} />);
        expect(result.queryByText("Test1")).toBeInTheDocument();
        expect(result.queryByText("Test2")).toBeInTheDocument();
        expect(result.queryAllByText("Edit").length).toEqual(2);
        expect(result.queryAllByText("Del").length).toEqual(2);
    });
});