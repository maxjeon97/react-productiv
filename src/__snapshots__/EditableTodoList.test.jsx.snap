// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditableTodoList component > matches snapshot 1`] = `
<div>
  <div
    class="EditableTodoList"
  >
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="1"
        >
          <div
            class="Todo-title"
          >
            <b>
              Test1
            </b>
            <small>
              (priority: 
              1
              )
            </small>
          </div>
          <div>
            <small>
              Test1 Descr
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="2"
        >
          <div
            class="Todo-title"
          >
            <b>
              Test2
            </b>
            <small>
              (priority: 
              2
              )
            </small>
          </div>
          <div>
            <small>
              Test2 Descr
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
